bitswap/client/internal/session/session.go

```diff
@@ -403,14 +403,20 @@ func (s *Session) handlePeriodicSearch(ctx context.Context) {
 // findMorePeers attempts to find more peers for a session by searching for
 // providers for the given Cid
 func (s *Session) findMorePeers(ctx context.Context, c cid.Cid) {
+	// noop when provider finder is disabled
+	if s.providerFinder == nil {
+		return
+	}
 	go func(k cid.Cid) {
 		ctx, span := internal.StartSpan(ctx, "Session.FindMorePeers")
 		defer span.End()
-		for p := range s.providerFinder.FindProvidersAsync(ctx, k) {
+		// Max is set to -1. This means "use the default limit" in the
+		// provider query manager.
+		for p := range s.providerFinder.FindProvidersAsync(ctx, k, -1) {
```

Comment:
It is a bit ugly that here it is assumed that the ContentDiscovery that we use understands what -1 means.

Reply:
ok, now calling it with 0.